cirq/pasqal/pasqal_device.py

```diff
@@ -77,21 +79,32 @@ def decompose_operation(self,
             raise TypeError("{!r} is not a gate operation.".format(operation))
 
         # Try to decompose the operation into elementary device operations
-        if not PasqalDevice.is_pasqal_device_op(operation):
-            decomposition = cirq.protocols.decompose(
-                operation, keep=PasqalDevice.is_pasqal_device_op)
+        if not self.is_pasqal_device_op(operation):
+            decomposition = PasqalConverter().pasqal_convert(
+                operation, keep=self.is_pasqal_device_op)
 
         return decomposition
 
-    @staticmethod
-    def is_pasqal_device_op(op: cirq.ops.Operation) -> bool:
+    def is_pasqal_device_op(self, op: cirq.ops.Operation) -> bool:
 
         if not isinstance(op, cirq.ops.Operation):
             raise ValueError('Got unknown operation:', op)
-        return (len(op.qubits) > 1) or isinstance(
-            op.gate, (cirq.ops.IdentityGate, cirq.ops.MeasurementGate,
-                      cirq.ops.PhasedXPowGate, cirq.ops.XPowGate,
-                      cirq.ops.YPowGate, cirq.ops.ZPowGate))
+
+        valid_op = isinstance(op.gate,
+                              (cirq.ops.IdentityGate, cirq.ops.MeasurementGate,
+                               cirq.ops.PhasedXPowGate, cirq.ops.XPowGate,
+                               cirq.ops.YPowGate, cirq.ops.ZPowGate))
+
+        if not valid_op:  # To prevent further checking if already passed
+            if isinstance(
+                    op.gate,
+                (cirq.ops.HPowGate, cirq.ops.CNotPowGate, cirq.ops.CZPowGate,
+                 cirq.ops.CCZPowGate, cirq.ops.CCXPowGate)):
+                expo = op.gate.exponent
+                if not isinstance(expo, sympy.Basic):
```

Comment:
1) Use `cirq.is_parametrized(op)` - that should take care of other parameters as well. 
2) Is the Pasqal device going to handle non-integer exponent and/or parametrized gates from PhasedXPowGate, XPowGate, YPowGate, ZPowGate?

Reply:
1. Ok, will do.
2. Yes, we'll handle non-integer exponents for these. As for the parametrization, we won't necessarily handle that, but we support the parameter substitution for actual values in `PasqalSampler.run_sweep`, so we have to allow those here.